integration-tests/tests/register-user.js

```diff
@@ -14,7 +14,7 @@ const testRegisterUserFn = withUser(async ({ page, testUrl, token }) => {
   await verifyAccessibility(page)
   await click(page, clickable({ textContains: 'Register' }))
   await click(page, clickable({ textContains: 'Accept' }), { timeout: 90000 })
-  await findText(page, 'To get started, Create a New Workspace')
+  await findText(page, 'Welcome to Terra Community Workbench')
```

Comment:
For some reason, this test was ending up on the workspaces page. But after clicking the accept button, Terra redirects to the home page.

https://github.com/DataBiosphere/terra-ui/blob/c0d416875ed08db77082d3f64393f0e783089d49/src/pages/TermsOfService.js#L58

Reply:
I believe it should end up on the workspaces pages because the log in process was initiated by clicking "View Workspaces" (line 10). So after registering, it should remember to go back to Workspaces.

Reply:
That was true before #3671, but that PR changed the behavior of the accept button from `Nav.history.goBack()` to `Nav.goToPath('root')`.

https://github.com/DataBiosphere/terra-ui/pull/3671/files#diff-6d4c9b0f2ed976516728fc98d614864ed93e78b692554d0e6a802d03fd60c890L49-R58

Reply:
Why was that change made?

Reply:
@Ghost-in-a-Jar can you answer this question?

Reply:
When testing this functionality across different environments we noticed that goBack would send us to whatever url we were browsing before upending terra. Sending us out of the platform. 

The rationale was to send users back to root instead of potentially outside of the platform. We are also react noobs though so I'm sure there is a better way to achieve the UX we want.

Reply:
I think it would definitely be better if `goBack` was used as long as it doesn't take you outside of Terra. I don't know off the top of my head how to do that.

But I'm curious-- what URL were you starting with that immediately required you to register? `https://app.terra.bio/` doesn't make you register immediately, so presumably one needs to click on something else in Terra to initiate the registration process.

Reply:
Oh, perhaps because you clicked log in vs. clicking on a page link that forced the log in.

Reply:
yeah that is a possibility -- i would definitely defer to UI folks though because Greg and I are total react noobs.

Reply:
I think we could probably code something fancy here as a followup if we want. Like use goBack to the last terra page in the session history -- if there arent any terra pages in the history just go to root.

Reply:
Agreed. In any case, `toToPath('root')` is currently in production, so I think changes to that can be a follow-up.